canard.c

```diff
@@ -957,6 +985,60 @@ CANARD_INTERNAL int16_t enqueueTxFrames(CanardInstance* ins,
     return result;
 }
 
+/** 
+ * Takes an unpadded transport frame and add padding if required.
+ * If input `data_len` is longer than max, max will be returned;
+ */
+CANARD_INTERNAL int16_t addRequiredPadding(CanardTransportFrame* frame)
+{
+    uint8_t data_len = frame->data_len;
+
+    uint8_t next_available_dlc;
+
+    switch (frame->protocol)
+    {
+        case CanardTransportProtocolCAN:
+        {
+            if (data_len <= 8)
+                next_available_dlc = data_len;
+            else
+                next_available_dlc = 8;            
+            break;
+        }
+        case CanardTransportProtocolFD:
+        {
+            if (data_len <= 8)
+                next_available_dlc = data_len;
+            else if (data_len <= 12)
+                next_available_dlc = 12;
+            else if (data_len <= 16)
+                next_available_dlc = 16;
+            else if (data_len <= 20)
+                next_available_dlc = 20;
+            else if (data_len <= 24)
+                next_available_dlc = 24;
+            else if (data_len <= 32)
+                next_available_dlc = 32;
+            else if (data_len <= 48)
+                next_available_dlc = 48;
+            else
+                next_available_dlc = 64;
+            break;
+        }
+        default:
+        {
+            return -CANARD_ERROR_INVALID_ARGUMENT;
+        }
+    }
+
+    for (int i = data_len; i <= next_available_dlc; i++)
```

Comment:
Please avoid reliance on the fundamental types (MISRA recommendation), use `uint8_t` instead.

Reply:
willfix, thanks